src/modules/flow/json/json.c

```diff
@@ -405,6 +405,22 @@ json_path_is_array_key(struct sol_str_slice slice)
     for (end_reason = SOL_JSON_LOOP_REASON_OK; \
         json_path_get_next_slice(&scanner, &key_slice, &end_reason);)
 
+static int32_t
+json_array_get_key_index(struct sol_str_slice key)
+{
+    long int index_val;
+    char *endptr;
+
+    errno = 0;
+    //TODO: use safe strtol from sol-util when available
+    index_val = strtol(key.data + 1, &endptr, 10);
+    if (endptr != key.data + key.len - 1 ||
+        index_val < 0 || errno > 0 || index_val > INT32_MAX)
+        return errno > 0 ? -errno : -EINVAL;
```

Comment:
Since `errno` is tested twice, just move that to a condition just for it: `if (errno) return -errno;`. No need to compare if it's greater than zero.